pkg/validation/config.go

```diff
@@ -395,33 +412,42 @@ var (
 	exceptions = sets.New[string]("openshift")
 )
 
-func validatePromotionConfiguration(fieldRoot string, input api.PromotionConfiguration, promotesOfficialImages, imageTargets bool, promotionNamespace string, releaseTagConfiguration *api.ReleaseTagConfiguration, releases map[string]api.UnresolvedRelease) []error {
+func validatePromotionConfiguration(ctx *configContext, input api.PromotionConfiguration, promotesOfficialImages, imageTargets bool, promotionNamespace string, releaseTagConfiguration *api.ReleaseTagConfiguration, releases map[string]api.UnresolvedRelease) []error {
 	var validationErrors []error
 
 	if len(input.Namespace) == 0 {
-		validationErrors = append(validationErrors, fmt.Errorf("%s: no namespace defined", fieldRoot))
+		validationErrors = append(validationErrors, ctx.errorf("no namespace defined"))
 	}
 
 	if openshiftWebhookForbiddingNamespaces.MatchString(input.Namespace) && !exceptions.Has(input.Namespace) {
-		validationErrors = append(validationErrors, fmt.Errorf("%s: cannot promote to namespace %s matching this regular expression: (^kube.*|^openshift.*|^default$|^redhat.*)", fieldRoot, input.Namespace))
+		validationErrors = append(validationErrors, ctx.errorf("cannot promote to namespace %s matching this regular expression: (^kube.*|^openshift.*|^default$|^redhat.*)", input.Namespace))
 	}
 
 	if len(input.Name) == 0 && len(input.Tag) == 0 {
-		validationErrors = append(validationErrors, fmt.Errorf("%s: no name or tag defined", fieldRoot))
+		validationErrors = append(validationErrors, ctx.errorf("no name or tag defined"))
 	}
 
 	if len(input.Name) != 0 && len(input.Tag) != 0 {
-		validationErrors = append(validationErrors, fmt.Errorf("%s: both name and tag defined", fieldRoot))
+		validationErrors = append(validationErrors, ctx.errorf("both name and tag defined"))
 	}
 
 	if promotesOfficialImages && imageTargets {
 		if _, ok := releases["latest"]; !ok && releaseTagConfiguration == nil {
-			validationErrors = append(validationErrors, fmt.Errorf("importing the release stream is required to ensure the promoted images to the namespace %s can be integrated properly. Although it can be achieved by tag_specification or releases[\"latest\"], adding an e2e test is strongly suggested", promotionNamespace))
+			validationErrors = append(validationErrors, ctx.errorf("importing the release stream is required to ensure the promoted images to the namespace %s can be integrated properly. Although it can be achieved by tag_specification or releases[\"latest\"], adding an e2e test is strongly suggested", promotionNamespace))
 		}
 	}
 	return validationErrors
 }
 
+func validateAccessoryPromotionConfiguration(ctx *configContext, input api.PromotionConfiguration) []error {
```

Comment:
Could we invalidate the cases where `AccessoryPromotionConfiguration` promotes to `ocp` or `okd` payload?
Those imagestreams have meaning for branching. Maybe excluding those cases to avoid questions such as "would it participate branching when it promotes to payload with `AccessoryPromotionConfiguration`"?
This should not be a really restriction for your cases the feature of this PR is all about promotion to other places than the payloads. ^_^

(If you agree, please include it in the GoDoc too. Thanks.)

Otherwise, the code looks good to me.

Reply:
Good call, done